### DISCUSSION TITLE
meta data in markdown files

### DISCUSSION THREAD

**Discussion by 4e576rt8uh9ij9okp** [markedjs/marked#3290] [Q&A]
Hi there,
I want to add some meta data into the markdown files maybe something like:

```
title: About Page
description: I have no friends
-----
# About Page
```

How could I do this?
So anything below the `-----` lines would be parsed but anything above wouldn't, it would be just for the sake of getting the meta data.

Or would it be better to just `.split("-----")` and then parse the meta data with my own function and pass the other half (markdown) to the marked parser?

**Comment by UziTech** [comment]
You can use the preprocess hook to parse front-matter before the markdown

```js
import { marked } from 'marked';
import fm from 'front-matter';

// Override function
const hooks = {
  preprocess(markdown) {
    const { attributes, body } = fm(markdown);
    for (const prop in attributes) {
      if (prop in this.options) {
        this.options[prop] = attributes[prop];
      }
    }
    return body;
  }
};

marked.use({ hooks });

// Run marked
console.log(marked.parse(`
---
headerIds: false
---

## test
`.trim()));
```

**Comment by jargoniseasy** [comment]
I simply go through with quick guide on stackoverflow [StackOverflow](https://stackoverflow.com/questions/44215896/markdown-metadata-format) and  used this MD type of meta data on Hashnode Blog [JargonIsEasy Blog](https://jargoniseasy.com/aravind-srinivas-perplexity-ceo-biography) canonical to dev.to [Canonical Dev.to Blog](https://dev.to/jargoniseasy/objects-in-javascript-3clp)
